### ISSUE TITLE
try different training scheduler

### ISSUE THREAD

**Issue by ArneBinder** [ArneBinder/pie-document-level#348]
It looks like the last part of the training is very beneficial, independent of the number of epochs  (e.g. see [this W&B project](https://wandb.ai/arne/dataset-sciarg-task-ner_re-v0.4-training)). This part may be characterized by small learning rate, but it is assumed, that the peak learning rate should not be changed. If this is the case, it may be beneficial to stretch this last training regime. This could be established by using a different scheduler such as provided by `get_cosine_schedule_with_warmup` from the `transformers` package (note that we usually use `get_linear_schedule_with_warmup`).

**Comment by ArneBinder** [comment]
See https://github.com/ArneBinder/pie-modules/pull/148 for the implementation.
